src/Symfony/Component/HttpFoundation/File/UploadedFile.php

```diff
@@ -251,8 +251,10 @@ public static function getMaxFilesize()
 
     /**
      * Returns the given size from an ini value in bytes.
+     *
+     * @return int|float Returns float if size > PHP_INT_MAX
```

Comment:
I've seen on php-internals that an alternative could be to use a string instead of a float.
I like that as it'd prevent loss of information. Would you mind giving it a try? (also for the implementation I mean)

Reply:
but that would be a big BC break as a string is not a number anymore and has to be manipulated in a dedicated way (if you just cast it to int or float after getting it, it does not prevent loss of information either)

Reply:
Sure, we should return a string only when an integer overflow would occur.
But that's the only way to make our API lossless. Returning a float wouldn't.

Reply:
but `int|string` is painful to use for our users, even more if the common case returns a `int` and so hides the `string` from their testing. Dealing with numbers representated as `string` requires a totally different approach.

Reply:
the thing is: what choice do we have?

Reply:
I would like to point out that the last time this was fixed (https://github.com/symfony/symfony/pull/34516) it was just changed to sometimes return a float (also there were no unit test changes required for that change). If you guys want to do something different, then I would think that it's on you to scan the entire codebase for where this can occur and make your decision on what the overall approach is going to be for handling this.

Reply:
Note that our own usage of this private method in `getMaxFilesize` already relies on doing math with it, so we would be affected ourselves by the issue of representing numbers as string.
So my vote would be to keep it as `int|float` (note that for PHP 64bit, you will not suffer from loss of information as you won't go over `PHP_INT_MAX` with your max upload size)